lib/oga/xml/parser.rll

```diff
@@ -245,7 +245,7 @@ string_body
   def reset
     @line = 1
 
-    @lexer.reset
+    @lexer.reset_native
   end
```

Comment:
@sferik Ah, I see the problem now. The class `Oga::XML::PullParser` also has a `reset` method that calls `super`. The code of said method can also be moved to `PullParser#initialize`, then the call to `reset` can be removed from `PullParser#parse` (the `parse` method itself here should stay).